addon/utils/ce/movement-observers/lump-node-movement-observer.js

```diff
@@ -1,43 +1,78 @@
+    if (isInLumpNode(newSelection.startNode.domNode, editor.rootNode)) {
       if (oldSelection && oldSelection.startNode.absolutePosition > newSelection.startNode.absolutePosition) {
         // seems a backward movement, set cursor before lump node
-        newNode = getPreviousNonLumpTextNode(newSelection.startNode.domNode, document.rootNode);
-        relativePosition = newNode.length;
+        jumpOverLumpNodeBackwards(newSelection.startNode.domNode, editor.rootNode, editor);
       }
       else {
         // seems a forward movement, set cursor after lump node
-        newNode = getNextNonLumpTextNode(newSelection.startNode.domNode, document.rootNode);
-        relativePosition = 0;
+        jumpOverLumpNode(newSelection.startNode.domNode, editor.rootNode, editor);
       }
-      document.updateRichNode();
-      document.setCaret(newNode, relativePosition);
     }
-    else if (isInLumpNode(newSelection.endNode.domNode, document.rootNode)) {
+    else if (isInLumpNode(newSelection.endNode.domNode, editor.rootNode)) {
       // startNode != endNode, a selection ending in a lumpNode, for now
       // for now just reset to start of the selection
-      document.setCaret(newSelection.startNode.domNode, newSelection.startNode.relativePosition);
+      editor.setCaret(newSelection.startNode.domNode, newSelection.startNode.relativePosition);
     }
   }
 }
+
+/**********************************************************************************************************
+ DISCLAIMER: this code is copypasta from a utils/plugins/lump-node/backspace-plugin.ts and will be revised.
+ It is meant to fix a bug where you could still click into a lump node in some cases.
+ Groundwork and collective thinking are need to deal with clicks and selecion changes (and arrows etc)
+ It will inject textNode if your previous or next sibling is an element. Not super elegant. Better than before.
+***********************************************************************************************************/
+function jumpOverLumpNode(node, rootNode, editor) {
+  const element = getParentLumpNode(node, rootNode);
+  let nextNode;
+  if(!element.nextSibling) {
+    nextNode = document.createTextNode('');
+    element.after(nextNode);
+  }
+  if(element.nextSibling && element.nextSibling.nodeType == Node.TEXT_NODE){
+    nextNode = element.nextSibling;
+  }
+  else {
+    nextNode = document.createTextNode('');
+    element.after(nextNode);
+  }
+  nextNode = ensureValidTextNodeForCaret(nextNode);
+  editor.updateRichNode();
+  moveCaret(nextNode, 0);
+}
+
+function jumpOverLumpNodeBackwards(node, rootNode, editor) {
+  const element = getParentLumpNode(node, rootNode);
+  let textNode;
+  if(element.previousSibling && element.previousSibling.nodeType == Node.TEXT_NODE){
+    textNode = element.previousSibling;
+  }
+  else {
+    textNode = document.createTextNode('');
+    element.before(textNode);
+  }
+  textNode = ensureValidTextNodeForCaret(textNode);
```

Comment:
there is no need to call ensureValidTextNodeForCaret, as setCaret should take care of that.

Reply:
I think there are still issue related to the implementation with this logic in setCaret.
 Mainly related to whitespace textnodes next to elements displayed as block and the caret not wanting to appear in there. Very visible if the lump node is the only node in the editor, and start clicking in it. Cursor will not follow (or display weirdly in FF). (Toggle the line, you'll see)

And setCaret relies heavily on richnode and fixing it there,might potentially break other stuff, I consider this as out of scope to merge both, and I suggest of wrapping this in a TODO. (I guess the approach will also change quite a bit with native handling and the virtual DOM limiting the possible cases)